code/datums/elements/dusts_on_leaving_area.dm

```diff
@@ -0,0 +1,28 @@
+/datum/element/dusts_on_leaving_area
+	element_flags = ELEMENT_DETACH | ELEMENT_BESPOKE
+	id_arg_index = 2
+	var/list/attached_mobs = list()
+	var/list/area_types = list()
+
+/datum/element/dusts_on_leaving_area/Attach(datum/target,types)
+	. = ..()
+	if(!ismob(target))
+		return ELEMENT_INCOMPATIBLE
+	attached_mobs += target
+	area_types = types
+	START_PROCESSING(SSprocessing,src)
+
+/datum/element/dusts_on_leaving_area/Detach(mob/M)
+	. = ..()
+	if(M in attached_mobs)
+		attached_mobs -= M
+	if(!attached_mobs.len)
+		STOP_PROCESSING(SSprocessing,src)
+
+/datum/element/dusts_on_leaving_area/process()
+	for(var/m in attached_mobs)
+		var/mob/M = m
+		var/area/A = get_area(M)
+		if(!(A.type in area_types))
+			M.dust(force = TRUE)
+			Detach(M)
```

Comment:
shouldn't this go before dust as i'm assuming elements auto detach on qdel?

also is it really worth elementing everything when you can just hook ghost cafe area to dust people with the ghost cafe roll or
feels really expensive for a niche  thing =|

Reply:
While, yea, while it should detach on Destoy(), keep in mind that elements are singletons assignable to multiple datums, and are only created if there is no other existing element that meets the criterias (mostly just the type, in certain cases arguments too), and thus lighter than components already.

Reply:
Yeah, this is a "just-in-case", and it won't cause any runtimes or issues AFAIK.